VocaDb.Migrations/Migrations.cs

```diff
@@ -41,18 +40,19 @@ public CommentTable(string name, string schema = null, string entryTypeName = nu
 
 		public override void Up()
 		{
+			// TODO: remove these columns
+			Alter.Column("AuthorName").OnTable(TableNames.AlbumComments).AsString(100).Nullable();
+			Alter.Column("AuthorName").OnTable(TableNames.DiscussionComments).InSchema(SchemaNames.Discussions).AsString(100).Nullable();
+			Alter.Column("Author").OnTable(TableNames.DiscussionTopics).InSchema(SchemaNames.Discussions).AsInt32().Nullable();
+			Alter.Column("Content").OnTable(TableNames.DiscussionTopics).InSchema(SchemaNames.Discussions).AsString(int.MaxValue).Nullable();
+
 			// Rename columns on `DiscussionTopics` table.
 			Rename.Column("Content").OnTable(TableNames.DiscussionTopics).InSchema(SchemaNames.Discussions).To("Message");
 
 			// Rename columns on `AlbumReviews` table.
 			Rename.Column("Date").OnTable(TableNames.AlbumReviews).To("Created");
 			Rename.Column("Text").OnTable(TableNames.AlbumReviews).To("Message");
 			Rename.Column("User").OnTable(TableNames.AlbumReviews).To("Author");
-			Delete.Index("UX_AlbumReviews").OnTable(TableNames.AlbumReviews);
-			Create.Index().OnTable(TableNames.AlbumReviews)
-				.OnColumn("Album").Ascending()
-				.OnColumn("Author").Ascending()
-				.OnColumn("LanguageCode").Ascending();

```

Comment:
Should I still add an index on the `Comments` table? Should it be `Album, Author, LanguageCode` or `Author, LanguageCode`?

Reply:
As a general rule, usually you want to add indexes to reference columns such as Album/Artist/Song, and also Author, because joins between database tables are more efficient with indexes. A separate index for each reference column of course, because they are generally not used together. You can add other columns, if those columns are generally used together with the reference columns.